builtin/logical/pki/path_roles.go

```diff
@@ -460,6 +494,7 @@ type roleEntry struct {
 	KeyUsage              string `json:"key_usage" structs:"key_usage" mapstructure:"key_usage"`
 	OU                    string `json:"ou" structs:"ou" mapstructure:"ou"`
 	Organization          string `json:"organization" structs:"organization" mapstructure:"organization"`
+	GenerateLease         *bool  `json:"generate_lease" structs:"generate_lease"`
```

Comment:
Easy fix: when you're using mapstructure to decode, when you allocate the struct for it to decode into, explicitly allocate the bool var.

However, I was thinking - the role entry doesn't have to match the API. So in the API we could do `generate_leases` but actually have the role entry value be "skip_lease_generation". That way rather than use a pointer, we can just have it be a normal `bool` and the upgrade path will work perfectly. On API call, just set the value to `!(data.Get("generate_leases").(bool))`.